components/async-operation-status-mgt/org.wso2.carbon.identity.framework.async.operation.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/operation/status/mgt/internal/dao/impl/AsyncOperationOperationStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,584 @@
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.SUBJECT_ID;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.SUBJECT_TYPE;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.TARGET_ORG_ID;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.constant.SQLConstants.SQLPlaceholders.UNIT_OPERATION_ID;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.constant.SQLConstants.UPDATE_ASYNC_OPERATION;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.util.AsyncOperationStatusMgtExceptionHandler.handleServerException;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.util.Utils.isMSSqlDB;
+import static org.wso2.carbon.identity.framework.async.operation.status.mgt.internal.util.Utils.isOracleDB;
+
+/**
+ * DAO implementation for Asynchronous Operation Status Management.
+ */
+public class AsyncOperationOperationStatusMgtDAOImpl implements AsyncOperationStatusMgtDAO {
+
+    private static final Log LOG = LogFactory.getLog(AsyncOperationOperationStatusMgtDAOImpl.class);
+
+    @Override
+    public String registerAsyncStatusWithoutUpdate(OperationInitDTO record) throws AsyncOperationStatusMgtException {
+
+        Timestamp currentTimestamp = new Timestamp(new Date().getTime());
+        String operationId = UUID.randomUUID().toString();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                     CREATE_ASYNC_OPERATION, OPERATION_ID)) {
+
+            statement.setString(OPERATION_ID, operationId);
+            statement.setString(CORRELATION_ID, record.getCorrelationId());
+            statement.setString(OPERATION_TYPE, record.getOperationType());
+            statement.setString(SUBJECT_TYPE, record.getOperationSubjectType());
+            statement.setString(SUBJECT_ID, record.getOperationSubjectId());
+            statement.setString(INITIATED_ORG_ID, record.getResidentOrgId());
+            statement.setString(INITIATED_USER_ID, record.getInitiatorId());
+            statement.setString(STATUS, OperationStatus.IN_PROGRESS.toString());
+            statement.setTimeStamp(CREATED_AT, currentTimestamp, null);
+            statement.setTimeStamp(LAST_MODIFIED, currentTimestamp, null);
+            statement.setString(POLICY, record.getOperationPolicy());
+            statement.executeUpdate();
+            return operationId;
+        } catch (SQLException e) {
+            throw handleServerException(ERROR_WHILE_PERSISTING_ASYNC_OPERATION_STATUS, e);
```

Comment:
Hope we can use `NamedJdbcTemplate` class for this as well. 
Let's be consistent across the class